langchain_bot.py

```diff
@@ -0,0 +1,66 @@
+from langchain.document_loaders import DirectoryLoader
+
+loader = DirectoryLoader(
+    './Langchain/data/', # my local directory
+    glob='**/*.pdf',     # we only get pdfs
+    show_progress=True
+)
+docs = loader.load()
+
+
+from langchain.text_splitter import CharacterTextSplitter
+
+text_splitter = CharacterTextSplitter(
+    chunk_size=1000, 
+    chunk_overlap=0
+)
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The code is loading and splitting documents without any error handling. If the directory does not exist or if there are no PDF files in the directory, this could lead to runtime errors. Consider adding error handling to ensure that the directory exists and contains PDF files before attempting to load and split them.
```diff
+import os
+import glob

...

+if os.path.exists('./Langchain/data/') and glob.glob('./Langchain/data/**/*.pdf'):
+    docs = loader.load()
+    docs_split = text_splitter.split_documents(docs)
+else:
+    print("Directory does not exist or no PDF files found.")
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->